app/controllers/security_ruleset_controller.rb

```diff
@@ -0,0 +1,41 @@
+
+class SecurityRulesetController < ApplicationController
+  include RulesetValidations
+
+  before_action :validate_file_upload!, only: [:create]
+  before_action :validate_ruleset_name!, only: [:new]
+  before_action :clear_errors
+
+  def new
+  end
+
+  def create
+    crunch_json = JSON.parse(crunch42_data)
+    @score = crunch_json['score']
+    @openapi = crunch_json['openapiState']
+    @counter = crunch_json['issueCounter']
+    @criticality = crunch_json['criticality']
+    @issues = crunch_json['data']['issues'].map do |issue|
+      lines = issue[1]['issues'].map { |occurrence| occurrence['pointer'] }
+      [issue[1]['criticality'],issue[1]['description'], lines]
```

Comment:
As I've stated previously, returning an array of arrays makes it difficult to work out from the view alone what is being displayed in each element. A hash would make this a lot clearer.

Reply:
We will build something that parses the data into a model, so this logic will have to move somewhere else. I'm not adding that in this pr though, I'll make what changes I can to improve the readability though, such as using pluck.